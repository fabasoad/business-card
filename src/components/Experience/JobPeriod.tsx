import * as React from 'react'
import { useTranslation } from 'react-i18next'
import DateLocale from '../controls/DateLocale'

interface JobPeriodProps {
  fromMonth: number
  fromYear: number
  toMonth?: number
  toYear?: number
}

export default function JobPeriod(props: JobPeriodProps) {
  const { t } = useTranslation()
  return (
    <div className="timeline-image">
      <h4>
        {(props.toMonth &&
          props.toYear &&
          <DateLocale month={props.toMonth} year={props.toYear} />) ||
          <div>{t('business-card-experience-present')}</div>}
        - <br />
        <DateLocale month={props.fromMonth} year={props.fromYear} />
      </h4>
    </div>
  )
}
