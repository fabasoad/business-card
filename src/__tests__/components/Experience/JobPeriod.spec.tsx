/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import JobPeriod from '../../../components/Experience/JobPeriod'

test('should render JobPeriod correctly', () => {
  const wrapper = shallow(<JobPeriod
    fromMonth={1}
    fromYear={2019}
    toMonth={4}
    toYear={2020}
  />)
  expect(wrapper).toMatchSnapshot()
})
