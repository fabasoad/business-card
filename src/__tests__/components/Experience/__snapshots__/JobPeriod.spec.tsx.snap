// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobPeriod correctly 1`] = `
<div
  className="timeline-image"
>
  <h4>
    <Connect(DateLocale)
      month={4}
      year={2020}
    />
    - 
    <br />
    <Connect(DateLocale)
      month={1}
      year={2019}
    />
  </h4>
</div>
`;
